### ISSUE TITLE
bug in ValidationPerformance coffee break?

### ISSUE THREAD

**Issue by jeff-regier** [pluskid/Mocha.jl#133]
I'd like to add a `ValidationPerformance` coffee break that computes the loss of the current model on the whole training set. (This is useful so I can monitor whether training loss and testing loss are diverging.) But when I modify `examples/mnist/mnist.jl`, replacing `test_net` with `net` on line 50, nothing gets printed any more:

```
24-Sep 15:13:50:INFO:root:
24-Sep 15:13:50:INFO:root:## Performance on Validation Set after 0 iterations
24-Sep 15:13:50:INFO:root:---------------------------------------------------------
24-Sep 15:13:50:INFO:root:---------------------------------------------------------
24-Sep 15:13:50:INFO:root:
```

Is this a bug? I got this output from Julia 0.3.11 on linux and a fresh installation of Mocha (master branch) with only line 50 of `mnist.jl` modified.

**Comment by pluskid** [comment]
@jeff-regier please don't use the net you are training to run validation, because running the net will increase the epoch and iteration counter of the data layer and screw up the whole solver training process. What should be perfectly fine is to construct another `net2` with the same parameters you used to construct `net`, and they should be able to run independent from each other.

**Comment by jeff-regier** [comment]
Got it, thanks!
